ts/main.ts

```diff
@@ -85,25 +94,46 @@ function viewSwap(viewName: string): void {
 }
 
 $photoUrl.addEventListener('input', (event: Event) => {
-  const target = event.target as HTMLInputElement;
-  $photoPreview.setAttribute('src', target.value);
+  const $eventTarget = event.target as HTMLInputElement;
+  $photoPreview.setAttribute('src', $eventTarget.value);
 });
 
 $entryForm.addEventListener('submit', (event: Event) => {
   event.preventDefault();
   const $formElements = $entryForm.elements as FormElements;
-  const entryObj: Entry = {
-    title: $formElements.title.value,
-    photo_url: $formElements.photo_url.value,
-    notes: $formElements.notes.value,
-    entryId: data.nextEntryId,
-  };
-  data.nextEntryId++;
-  data.entries.unshift(entryObj);
-  $photoPreview.setAttribute('src', 'images/placeholder-image-square.jpg');
+
+  if (data.editing === null) {
+    const newEntryObj: Entry = {
+      title: $formElements.title.value,
+      photo_url: $formElements.photo_url.value,
+      notes: $formElements.notes.value,
+      entryId: data.nextEntryId,
+    };
+    data.nextEntryId++;
+    data.entries.unshift(newEntryObj);
+    $photoPreview.setAttribute('src', 'images/placeholder-image-square.jpg');
+    $entriesUl.prepend(renderEntry(newEntryObj));
+  }
+
+  if (data.editing !== null) {
+    const editEntryObj: Entry = {
+      title: $formElements.title.value,
+      photo_url: $formElements.photo_url.value,
+      notes: $formElements.notes.value,
+      entryId: data.editing.entryId,
+    };
+    data.entries[data.entries.length - data.editing.entryId] = editEntryObj;
```

Comment:
This won't always work, because `entryId` won't get reset. The next feature is going to consist of deleting an entry, and so the length could be something like 2 but the entryId could be something like 500.

I would find a different way of finding this element